api/src/main/java/org/hawkular/alerts/api/model/trigger/Trigger.java

```diff
@@ -263,6 +253,40 @@ public Trigger(String tenantId, String id, String name, Map<String, String> cont
         this(tenantId, id, name, context, null);
     }
 
+    public Trigger(Trigger trigger) {
+        if (trigger == null) {
+            throw new IllegalArgumentException("trigger must be not null");
+        }
+        this.tenantId = trigger.getTenantId();
+        this.id = trigger.getId();
+        this.name = trigger.getName();
+        this.context = new HashMap<>(trigger.getContext());
+        this.tags = new HashMap<>(trigger.getTags());
+        this.actions = new HashSet<>();
+        for (TriggerAction action : trigger.getActions()) {
+            this.actions.add(new TriggerAction(action));
+        }
+        this.autoDisable = trigger.isAutoDisable();
+        this.autoEnable = trigger.isAutoEnable();
+        this.autoResolve = trigger.isAutoResolve();
+        this.autoResolveAlerts = trigger.isAutoResolveAlerts();
+        this.autoResolveMatch = trigger.getAutoResolveMatch();
+        this.dataIdMap = new HashMap<>(trigger.getDataIdMap() != null ? trigger.getDataIdMap() : new HashMap<>());
```

Comment:
Minor, I think you meant:
` this.dataIdMap = trigger.getDataIdMap() != null ? new HashMap<>(trigger.getDataIdMap()) : new HashMap<>());`